pkg/host/internal/infiniband/infiniband_test.go

```diff
@@ -0,0 +1,44 @@
+package infiniband
+
+import (
+	"fmt"
+	"net"
+
+	"github.com/golang/mock/gomock"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	netlinkLibPkg "github.com/k8snetworkplumbingwg/sriov-network-operator/pkg/host/internal/lib/netlink"
+	netlinkMockPkg "github.com/k8snetworkplumbingwg/sriov-network-operator/pkg/host/internal/lib/netlink/mock"
+	hostMockPkg "github.com/k8snetworkplumbingwg/sriov-network-operator/pkg/host/mock"
+)
+
+var _ = Describe("infiniband interface implementation", func() {
+	It("should create infiniband helper if guid config path is empty", func() {
+		testCtrl := gomock.NewController(GinkgoT())
+		netlinkLibMock := netlinkMockPkg.NewMockNetlinkLib(testCtrl)
+		netlinkLibMock.EXPECT().LinkList().Return([]netlinkLibPkg.Link{}, nil).AnyTimes()
+
+		_, err := New(netlinkLibMock, nil, nil)
+		Expect(err).NotTo(HaveOccurred())
+	})
+	It("should assign guids if guid pool is nil", func() {
+		testCtrl := gomock.NewController(GinkgoT())
+		netlinkLibMock := netlinkMockPkg.NewMockNetlinkLib(testCtrl)
+		netlinkLibMock.EXPECT().LinkList().Return([]netlinkLibPkg.Link{}, nil).AnyTimes()
+		netlinkLibMock.EXPECT().LinkSetVfNodeGUID(gomock.Any(), gomock.Any(), gomock.Any()).DoAndReturn(func(link netlinkLibPkg.Link, vf int, nodeguid net.HardwareAddr) error {
+			return fmt.Errorf(nodeguid.String())
+		}).Times(1)
+		netlinkLibMock.EXPECT().LinkSetVfPortGUID(gomock.Any(), gomock.Any(), gomock.Any()).Return(nil).Times(1)
+		kernelHelperMock := hostMockPkg.NewMockHostManagerInterface(testCtrl)
+		kernelHelperMock.EXPECT().Unbind(gomock.Any()).Return(nil).Times(1)
+
+		ib, err := New(netlinkLibMock, nil, nil)
+		Expect(err).NotTo(HaveOccurred())
+
+		err = ib.ConfigureVfGUID("randomAddr", "randomAddr", 0, nil)
+		Expect(err).To(HaveOccurred())
+		_, err = ParseGUID(err.Error())
```

Comment:
I don't understand usage of err.Error() at this line

Reply:
I added a comment on that. Essentially the mocked function returns a error with the GUID as a message so that we can validate, that a valid GUID was assigned in the absence of guid pool